src/rollup/index.ts

```diff
@@ -147,14 +148,44 @@ export function setWatcher(watcher: RollupWatcher) {
 	curWatcher = watcher;
 }
 
+function augmentChunkHashes(graph: Graph, chunks: Chunk[]): void {
+	for (let i = 0; i < chunks.length; i++) {
+		const chunk = chunks[i];
+		const facadeModule = chunk.facadeModule;
+		const preRenderedChunk = {
+			dynamicImports: chunk.getDynamicImportIds(),
+			exports: chunk.getExportNames(),
+			facadeModuleId: facadeModule && facadeModule.id,
+			imports: chunk.getImportIds(),
+			isDynamicEntry: facadeModule !== null && facadeModule.dynamicallyImportedBy.length > 0,
+			isEntry: facadeModule !== null && facadeModule.isEntryPoint,
+			modules: chunk.renderedModules,
+			get name() {
+				return chunk.getChunkName();
+			}
+		} as PreRenderedChunk;
+		const augmentedHash = graph.pluginDriver.hookReduceValueSync(
+			'augmentChunkHash',
+			'',
+			[preRenderedChunk],
+			(hashForChunk, pluginHash) => {
+				if (pluginHash) {
+					hashForChunk += pluginHash;
+				}
+				return hashForChunk;
+			}
+		);
+		chunk.augmentedHash = augmentedHash;
+	}
+}
+
 function assignChunksToBundle(
 	chunks: Chunk[],
 	outputBundle: OutputBundleWithPlaceholders
 ): OutputBundle {
 	for (let i = 0; i < chunks.length; i++) {
 		const chunk = chunks[i];
 		const facadeModule = chunk.facadeModule;
-
 		outputBundle[chunk.id as string] = {
```

Comment:
In the end I ended up duplicating this info since a great part of it must be recalculated since internal Ids change from `augmentChunkHashes` to `assignChunksToBundle` and I thought it would be messier to keep track of what should/shouldn't be updated.